FreeRTOS/Test/CMock/tasks/Makefile

```diff
@@ -0,0 +1,172 @@
+# Change according to what your  unit test directory is.
+# For example if testing queue.c your directory should be called queue
+# and the project name should be queue
+# if testing list.c your directory should be called list
+# and the project name should be list
+
+PROJECT          :=   tasks
+
+# List the dependency files you wish to mock
+MOCK_FILES_FP    :=   $(KERNEL_DIR)/include/timers.h
+MOCK_FILES_FP    +=   $(KERNEL_DIR)/include/list.h
+MOCK_FILES_FP    +=   $(KERNEL_DIR)/include/portable.h
+
+# List special compilation flags for this module
+CFLAGS += -Wno-error=pointer-to-int-cast -Wno-pointer-to-int-cast
+CFLAGS += -Wno-error=int-to-pointer-cast -Wno-int-to-pointer-cast
+
+# List special preprocessing flags for this module
+CPPFLAGS         +=
+
+# List special linker flags for this module
+LDFLAGS          +=   -Wl,--allow-multiple-definition 
+
+# Makefile Flags (if needed)
+#MAKEFLAGS += -rR
+
+# Try not to edit beyond this line
+
+COVERAGE_OPTS    :=   -fprofile-arcs -ftest-coverage -fprofile-generate
+
+# build/generated/queue
+SCRATCH_DIR      :=   $(GENERATED_DIR)/$(PROJECT)
+
+# list.h   timers.h   portable.h
+MOCK_FILES       :=   $(notdir $(MOCK_FILES_FP))
+
+# ...tasks/%/mocks/mock_list.o   tasks/%/mocks/mock_timers.o
+MOCK_OBJS        :=   $(addprefix $(SCRATCH_DIR)/%/mocks/mock_,$(MOCK_FILES:.h=.o))
+
+# ...tasks/%/mocks/mock_list.c   tasks/%/mocks/mock_timers.c
+MOCK_SRC_LIST    :=   $(addprefix $(SCRATCH_DIR)/%/mocks/,$(addprefix mock_,$(MOCK_FILES:.h=.c)))
+
+# executable generic name queue_utest without the _num
+EXEC             :=   $(PROJECT)_utest
+
+# Test/CMock/tasks
+PROJECT_DIR      :=   $(abspath .)
+
+# tasks/%/cpp/timers.h   tasks/%/cpp/list.h
+CPP_FILES        :=   $(addprefix $(SCRATCH_DIR)/%/cpp/,$(notdir $(MOCK_FILES_FP)))
+
+# tasks_utest_1.c   tasks_utest_2.c ...
+SOURCES          :=   $(sort $(wildcard $(PROJECT)_utest_*.c))
+
+# 1 2 3...
+PROJECT_NUMBERS  :=   $(subst .c,,$(subst $(PROJECT)_utest_,,$(SOURCES)))
+
+# tasks/1/mocks   tasks/2/mocks
+MOCK_DIRS        :=   $(addsuffix /mocks,$(addprefix $(SCRATCH_DIR)/,$(PROJECT_NUMBERS)))
+# queue/1/cpp   queue/2/cpp
+CPP_DIRS         :=   $(addsuffix /cpp,$(addprefix $(SCRATCH_DIR)/,$(PROJECT_NUMBERS)))
+# tasks/1/include   tasks/2/include
+INCLUDE_DIRS     :=   $(addsuffix /include,$(addprefix $(SCRATCH_DIR)/,$(PROJECT_NUMBERS)))
+#list/1   list/2 ....
+CONFIG_DIRS := $(addprefix $(SCRATCH_DIR)/,$(PROJECT_NUMBERS))
+
+ifeq ($(MOCK_FILES_FP),)
+     $(shell mkdir -p  $(MOCK_DIRS))
+     $(shell for n in $(MOCK_DIRS) ; do touch -a $(n)/mock_dummy.c)
+endif
+
+EXECUTABLES := $(addprefix $(BIN_DIR)/,$(patsubst %.c,%,$(SOURCES)))
+EXEC_PATH := $(addprefix $(BIN_DIR)/,$(EXEC))
+
+.PHONY: all Makefile directories
+
+.SECONDEXPANSION:
```

Comment:
I'm not familiar with how SECONDEXPANSION / SECONDARY works. It seems to build correctly though (once I replace mkdir with mkdir -p).

Reply:
it is not currently building without mkdir -p ?
Secondary expansion is used to be able to use make auto variables in the requirements

Reply:
Yes. I had quite a bit of trouble getting this branch to build locally without hacking up the Makefiles.

Reply:
can you incorporate the changes to the makefile ? or is it just the -p in makedir ?